pkg/apis/servicecatalog/types.go

```diff
@@ -774,6 +774,10 @@ type ServiceInstanceSpec struct {
 	// allows for parameters to be updated with any out-of-band changes that have
 	// been made to the secrets from which the parameters are sourced.
 	UpdateRequests int64
+
+	// DeletionRequested provides an imperative means to request
+	// deletion by doing an update.
+	DeletionRequested bool
```

Comment:
I do not think this matches with the rest of kubernetes nor does it mesh with the norms of OSB. I would suggest:

`unbindTimestamp time.Time`

Then you can set the time the unbind was requested. That is declarative.

Reply:
And then roll back by blanking it?

Reply:
it would work the same way as your bool, but the date matches with how k8s does deletes already